tutorials/intro/plot_introduction.py

```diff
@@ -1,404 +1,293 @@
-
-First, load the mne package:
-"""
-
-import mne
 
-##############################################################################
-# If you'd like to turn information status messages off:
+Introduction to  MEG and EEG data processing with MNE-Python
+============================================================
 
-mne.set_log_level('WARNING')
-
-##############################################################################
-# But it's generally a good idea to leave them on:
+This tutorial covers the basic EEG/MEG pipeline for event-related analysis:
+loading data, epoching, averaging, plotting, and estimating cortical activity
+from sensor data. It introduces the core MNE-Python data structures
+:class:`~mne.io.Raw`, :class:`~mne.Epochs`, :class:`~mne.Evoked`, and
+:class:`~mne.SourceEstimate`, and covers a lot of ground fairly quickly (at the
+expense of depth). Subsequent tutorials address each of these topics in greater
+detail. We begin by importing the necessary Python modules:
+"""
 
-mne.set_log_level('INFO')
+import os
+import numpy as np
+import mne
 
-##############################################################################
-# You can set the default level in every session by setting the environment
-# variable "MNE_LOGGING_LEVEL", or by having mne-python write preferences to a
-# file with::
+###############################################################################
+# Loading data
+# ^^^^^^^^^^^^
 #
-#     >>> mne.set_config('MNE_LOGGING_LEVEL', 'WARNING')
+# MNE-Python data structures are based around the FIF file format from
+# Neuromag, but there are reader functions for :ref:`a wide variety of other
+# data formats <data-formats>`. MNE-Python also has interfaces to a
+# variety of :doc:`publicly available datasets <../../manual/datasets_index>`,
+# which MNE-Python can download and manage for you.
 #
-# Note that the location of the mne-python preferences file (for easier manual
-# editing) can be found using:
-
-print(mne.get_config_path())
-
-##############################################################################
-# By default logging messages print to the console, but look at
-# :func:`mne.set_log_file` to save output to a file.
+# We'll start this tutorial by loading one of the example datasets (called
+# ":ref:`sample-dataset`"), which contains EEG and MEG data from one subject
+# performing an audiovisual experiment, along with structural MRI scans for
+# that subject. The :func:`mne.datasets.sample.data_path` function will
+# automatically download the dataset if it isn't found in one of the expected
+# locations, then return the directory path to the dataset (see the
+# documentation of :func:`~mne.datasets.sample.data_path` for a list of places
+# it checks before downloading).
+
+sample_data_folder = mne.datasets.sample.data_path()
+sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
+                                    'sample_audvis_raw.fif')
```

Comment:
I would actually not bother using os.path.join. If you just say:

```
raw_fname = sample_data_folder + '/MEG/sample/sample_audvis_raw.fif'
```
I think this should work in most machines (and it's what we do in many examples). It will keep the tutorial simple so that people are less bogged down by minutiae of how Python works

Reply:
strong disagree.  It is not always obvious to beginners what should happen when "adding" two strings, and windows users who are used to slashes going the other way might linger over your version of the line.  `os.path.join` is useful scaffolding when reading code as a beginner: "OK, joining paths here, nothing fancy, moving on".  This is also precisely why I'm **not** using `op.join` --- to keep the word "path" in there.

Reply:
okay fair enough!